From 5f623b0eb3f16f71b8024756c46ed6315651b618 Mon Sep 17 00:00:00 2001
From: gbhakteX <gurudattax.bhakte@intel.com>
Date: Thu, 18 Apr 2013 14:35:18 +0530
Subject: [PORT FROM MAIN]Incompatibility in class IAudioFlinger, in libmedia.so causing certain third-party-apps not to work.

BZ: 81522

In class IAudioFlinger, virtual functions added to support Offload (getMode(),
isAudioEffectEnabled() and getOffloadBufferSize()), caused incompatibility in binary.
With this patch providing a temporary solution, where the new added functions are
moved to the end of the class to make VT compatible.

Orig-Change-Id: Iba4eb831e86892da12541f7e32852e31b1f13d11
Signed-off-by: gbhakteX <gurudattax.bhakte@intel.com>
Signed-off-by: Santosh <santoshx.joshi@intel>

Change-Id: I3115c1570eaca2fa6581a25d34734416341e4d55
Signed-off-by: Arulselvan M <arulselvan.m@intel.com>
---
 include/media/IAudioFlinger.h |   11 ++++++-----
 1 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/include/media/IAudioFlinger.h b/include/media/IAudioFlinger.h
index 2b578f4..d92925f 100644
--- a/include/media/IAudioFlinger.h
+++ b/include/media/IAudioFlinger.h
@@ -115,8 +115,6 @@ public:
 
     // set audio mode
     virtual     status_t    setMode(audio_mode_t mode) = 0;
-    // get audio mode
-    virtual audio_mode_t   getMode() const  = 0;
 
     // mic mute/state
     virtual     status_t    setMicMute(bool state) = 0;
@@ -193,8 +191,12 @@ public:
     // helpers for android.media.AudioManager.getProperty(), see description there for meaning
     // FIXME move these APIs to AudioPolicy to permit a more accurate implementation
     // that looks on primary device for a stream with fast flag, primary flag, or first one.
-    virtual uint32_t getPrimaryOutputSamplingRate() = 0;
-    virtual size_t getPrimaryOutputFrameCount() = 0;
+    virtual int32_t getPrimaryOutputSamplingRate() = 0;
+    virtual int32_t getPrimaryOutputFrameCount() = 0;
+
+    // get audio mode
+    virtual audio_mode_t   getMode() const  = 0;
+
     virtual bool isAudioEffectEnabled(int sessionId) const = 0;
 
     virtual size_t getOffloadBufferSize(
@@ -202,7 +204,6 @@ public:
             uint32_t sampleRate,
             uint32_t channel,
             int output) = 0;
-
 };
 
 
-- 
1.7.4.1

